python/ray/data/datasource/parquet_datasource.py

```diff
@@ -144,7 +155,7 @@ def _deserialize_fragments_with_retry(
 
 
 @PublicAPI
-class ParquetDatasource(ParquetBaseDatasource):
+class ParquetDatasource(Datasource):
```

Comment:
For my own understanding, why doesn't this inherit from ParquetBaseDatasource anymore?

Reply:
`ParquetDatasource` was inheriting from `ParquetBaseDatasource` to get the write-related logic. Since the write logic has been moved to `ParquetDatasink`, there's not point for this inheritance anymore.

The inheritance was jank to begin with anyway. They should really be two separate classes like `ParquetDatasource` and `BulkParquetDatasource`

Reply:
Ah okay great, thanks!